spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/FunctionConfiguration.java

```diff
@@ -484,7 +485,17 @@ private void bindFunctionToDestinations(BindableProxyFactory bindableProxyFactor
 								+ "consumer, given that project reactor maintains its own concurrency mechanism. Was '..."
 								+ inputBindingName + ".consumer.concurrency=" + consumerProperties.getConcurrency() + "'");
 					}
-					SubscribableChannel inputChannel = this.applicationContext.getBean(inputBindingName, SubscribableChannel.class);
+					MessageChannel inputChannel = null;
+					final String reactive = environment.getProperty("spring.cloud.stream.reactive");
+					if (StringUtils.hasText(reactive)) {
```

Comment:
We can simplify this just into `Boolean.parseBoolean()`. If it is `null` or does not have value, the `false` is going to be returned.